Clinical-Genomics/scout
# Fix filters parse and display

This PR adds a functionality or fixes a bug.

- Fix #5060 

Bit of a difference already without touching parsing:
![Screenshot 2024-11-22 at 10 44 30](https://github.com/user-attachments/assets/6a610836-0a8e-416b-b25e-0320d8aec2f0)

Then with parsing fixed:
![Screenshot 2024-11-22 at 14 21 35](https://github.com/user-attachments/assets/2fde62ad-16c0-486e-870a-c2dfd5d13486)

![Screenshot 2024-11-22 at 14 34 12](https://github.com/user-attachments/assets/47a04240-cc71-4ab5-86da-3d81f7411bd8)

I also added a filters badge to the variant page for good measure:
![Screenshot 2024-11-22 at 14 49 06](https://github.com/user-attachments/assets/b139b908-9b20-4069-a935-a4996c570ad0)


<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>


**How to test**:
1. check a cancer case and an RD case, note filter tags missing and tags saying Pass when they should say Filtered. 
2. apply patch
3. note improvements with old cases, e.g. some new danger badges on variantS, and the filters badge on the variant page. And in particular no new issues with them.
4. for full effect, load/update the cases with the patch and note good tags both for callers and filters

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
